tests/test_framework/utils.py

```diff
@@ -457,6 +457,8 @@ def wait_for_logs(self, regexs, timeout=TIMEOUT):
 
             with self.logs_cond:
                 if pos >= len(self.logs):
+                    if not self.running:
+                        raise ValueError("Process died while waiting for logs")
```

Comment:
But, what if the process dies without printing the logs you were looking for ?

Reply:
This is checking if the process is dead when it's done reading the logs. So if I don't find the logs I was looking for I'm going to read all of them till the last one, check if the process is dead, if it is raise an error

Reply:
The point is that if you have 10 lines of logs you have to read all of them before saying "Process died while waiting for logs".
An example: we're looking for "Stopping revaultd", which is located in the logs at line 8
Python is slow at processing logs, when it's processing line 5 it notices that the process is dead and raises an error (happens to me every time in local)
This is wrong, because the log is there. Python should process every log and just then raise an error if the daemon is dead

Reply:
Oh, right. Makes sense. But iirc i had a patch for this which actually made this check an `elif` of:
```python
            if timeout is not None and time.time() > start_time + timeout:
                print("Time-out: can't find {} in logs".format(exs))
                for r in exs:
                    if self.is_in_log(r):
                        print("({} was previously in logs!)".format(r))
                 raise TimeoutError('Unable to find "{}" in logs.'.format(exs))
```
in #133 ? Maybe not. Anyways this or checking just above looks good, let's see what CI says with timeouts.

Reply:
I cherry-picked this commit from #133 as I expect to merge this one before and this is not really related to the json rpc thread anyway 
There is another commit in #133 that helps with tests: https://github.com/revault/revaultd/pull/133/commits/291c60d0c0d539782f38ff9cf93020329f029800
Can't find anything else